### PR TITLE
[Aviraj] Added blog how-to-create-an-nft-with-royalties-on-celo.md

### PR DESCRIPTION
**Pull request by avirajkhare00**

### CODE REVIEW

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+
+Returns true if this contract implements the interface defined by `interfaceId`.
+
+```js
+function _burn(uint256 tokenId) internal virtual override {
+    super._burn(tokenId);
+    _resetTokenRoyalty(tokenId);
+  }
+
+  function burnNFT(uint256 tokenId)
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:195]:
```suggestion
function burnNFT(uint256 tokenId)
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+      _tokenIds.increment();
+
+      uint256 newItemId = _tokenIds.current();
+      _safeMint(recipient, newItemId);
+      _setTokenURI(newItemId, tokenURI);
+
+      return newItemId;
+  }
+
+  function mintNFTWithRoyalty(address recipient, string memory tokenURI, address royaltyReceiver, uint96 feeNumerator)
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:216]:
```suggestion
function mintNFTWithRoyalty(address recipient, string memory tokenURI, address royaltyReceiver, uint96 feeNumerator)
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+      uint256 tokenId = mintNFT(recipient, tokenURI);
+      _setTokenRoyalty(tokenId, royaltyReceiver, feeNumerator);
+
+      return tokenId;
+  }
+```
+
+Here we have two functions. In our NFTMinter smart contract, we will call `mintNFTWithRoyalty()` function. This function takes `recipient`, `tokenURI`, `royaltyReceiver` and `feeNumerator`.
+
+Here, `recipient` takes an address whose NFT we need to mint.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:228]:
```suggestion
- `recipient` takes an address whose NFT we need to mint.
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+`tokenURI` contains IPFS hash of an image.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:229]:
```suggestion
- `tokenURI` contains IPFS hash of an image.
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+`royaltyReceiver` takes an address where we do need to send the royalty.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:230]:
```suggestion
- `royaltyReceiver` takes an address where we do need to send the royalty.
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+`feeNumerator` takes an integer, here we decide how much royalty do we need to send to the creator.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:231]:
```suggestion
- `feeNumerator` takes an integer, here we decide how much royalty do we need to send to the creator.
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+Here, we are creating `NFTMinter` contract.
+
+```js
+address public owner;
+uint256 public basePrice;
+address public MyNFTContractAddress;
+```
+
+We have defined three variables here.
+`owner`: holds owner address. One who instantiate smart contract becomes an owner.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:288]:
```suggestion
- `owner`: holds owner address. One who instantiate smart contract becomes an owner.
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+`basePrice`: Price at which we are minting the NFT.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:289]:
```suggestion
- `basePrice`: Price at which we are minting the NFT.
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+`MyNFTContractAddress`: It holds contract address of `MyNFT` smart contract.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:290]:
```suggestion
- `MyNFTContractAddress`: It holds contract address of `MyNFT` smart contract.
```

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+
+Since Rarible does not support Celo Testnet, I will be using goreli testnet to show you how royalty is transferred.
+
+Go ahead and run the frontend code by using any http server. Mint an NFT, then go to Rarible testnet here: https://testnet.rarible.com/.
+
+You can see that your NFT is getting displayed, now put it on sale for whatever amount you wish for.
+
+Now ask your friend or choose another wallet and purchase the NFT.
+
+Check for Etherscan transactions. You will see that `0.001` ether went to the address who minted NFT and 2.5 % went to the creater of the NFT.
```
Comment (viral-sangani) [inline blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md:363]: 2.5% or 2% royalty?

**Review by viral-sangani — CHANGES_REQUESTED** [review]

**Comment by viral-sangani** [comment]
Hey @avirajkhare00 Just made few changes, please look into it.
Greate tutorial btw, keep it up.

**Comment by viral-sangani** [comment]
Did you add this tutorial in `users.tsx` file?

File: blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md (hunk 10 above)
Comment (avirajkhare00) [reply blog/2022-12-22-how-to-create-an-nft-with-royalties-on-celo.md]: It's 2%. I have fixed it. Thanks for pointing it out

**Comment by avirajkhare00** [comment]
Hey @viral-sangani, I have accepted all the changes. Do you mean adding the tutorial inside `src/data-tutorials/users.tsx` file?

**Review by viral-sangani — APPROVED** [review]

**Pull request merged**